File: packages/sdk-ts/src/extensions/helix/layers.ts
/**
 * DEPRECATED shim — DefaultLayerResolver moved into the kernel.
 *
 * Layer resolution is a core kernel responsibility
 * (s-invert-layer-resolver-dep, 2026-07-07): the kernel must work with
 * zero extensions loaded, so the resolver now lives at
 * `src/kernel/layer-resolver.ts`. This module reexports the old public
 * names for external importers and emits a DeprecationWarning on import.
 *
 * @deprecated Import from `dna-sdk/kernel/layer-resolver` instead.
 */

export {
  DefaultLayerResolver,
  deepMerge,
  mergeTimelineArrays,
  type LayerSource,
} from "../../kernel/layer-resolver.js";

// Node/Bun idiomatic import-time deprecation (parity with the Python
// shim's warnings.warn(DeprecationWarning)).
if (typeof process !== "undefined" && typeof process.emitWarning === "function") {
  process.emitWarning(
    "extensions/helix/layers is deprecated — import DefaultLayerResolver/" +
      "deepMerge from kernel/layer-resolver instead (s-invert-layer-resolver-dep).",
    "DeprecationWarning",
  );
}

File: packages/sdk-py/dna/extensions/helix/layers.py
"""DEPRECATED shim — DefaultLayerResolver moved into the kernel.

Layer resolution is a core kernel responsibility
(s-invert-layer-resolver-dep, 2026-07-07): the kernel must work with
zero extensions loaded, so the resolver now lives at
``dna.kernel.compose.layer_resolver``. This module reexports the old
public names for external importers and warns on import.
"""
from __future__ import annotations

import warnings

from dna.kernel.compose.layer_resolver import (  # noqa: F401
    DefaultLayerResolver,
    _merge_timeline_arrays,
    _stamp_overlay_metadata,
    deep_merge,
)

warnings.warn(
    "dna.extensions.helix.layers is deprecated — import "
    "DefaultLayerResolver/deep_merge from dna.kernel.compose.layer_resolver "
    "instead (s-invert-layer-resolver-dep).",
    DeprecationWarning,
    stacklevel=2,
)

__all__ = [
    "DefaultLayerResolver",
    "deep_merge",
    "_merge_timeline_arrays",
    "_stamp_overlay_metadata",
]
